components/core/src/clp/ffi/ir_stream/search/QueryHandler.hpp

```diff
@@ -0,0 +1,111 @@
+#ifndef CLP_FFI_IR_STREAM_SEARCH_QUERYHANDLER_HPP
+#define CLP_FFI_IR_STREAM_SEARCH_QUERYHANDLER_HPP
+
+#include <outcome/outcome.hpp>
+
+#include "../../KeyValuePairLogEvent.hpp"
+#include "../../SchemaTree.hpp"
+#include "AstEvaluationResult.hpp"
+#include "ErrorCode.hpp"
+#include "NewProjectedSchemaTreeNodeCallbackReq.hpp"
+
+namespace clp::ffi::ir_stream::search {
+/**
+ * Class for handing KV-pair IR stream search queries.
+ *
+ * Each query handler stores a KQL query represented as an AST. The handler is responsible for:
+ * - resolving column descriptors to concrete schema tree nodes within the stream.
```

Comment:
```suggestion
 * - resolving column descriptors for schema-tree nodes within the stream.
```

Unless "concrete" has a specific meaning here?